biblib/views/operations_view.py

```diff
@@ -238,6 +216,32 @@ def post(self, library):
                         return err(BAD_LIBRARY_ID_ERROR)
                     secondary = session.query(Library).filter_by(id=secondary_uuid).one()
                     lib_names.append(secondary.name)
+                    check_access.append(secondary)
+
+            if data['action'] in ['union', 'intersection', 'difference']:
+                if 'libraries' not in data:
+                    return err(NO_LIBRARY_SPECIFIED_ERROR)
+                for lib in check_access:
+                    lib = session.merge(lib)
+                    if lib.public or not self.read_access(service_uid=user_editing_uid,
```

Comment:
Whoops my comment wasn't very clear - this should be `if not lib.public or not self.read_access...`